src/NLog/Config/ServiceRepository.cs

```diff
@@ -95,30 +97,70 @@ private object DefaultResolveInstance(Type itemType, HashSet<Type> seenTypes)
                 return objectResolver(itemType);
             }
 
-            try
+            if (_lateBoundMap.TryGetValue(itemType, out var compiledConstructor))
```

Comment:
Need try catch here with fallback to lookup with lock.

Reply:
done :)

please double check, thanks!